### ISSUE TITLE
Investigate memory limits set by scheduler in Flight

### ISSUE THREAD

**Issue by wdm78** [alces-software/flight-aws-marketplace#71]
Need to check that memory limits set on each compute node are appropriate to all instance types - they should ideally be close to (but not over) the amount of memory the node actually has.

**Comment by vlj91** [comment]
The memory limit for each instance type available through the MP templates appears to have roughly half (sometimes more or less) of its total memory in the scheduler total memory input. Oddly a `c4.large` has slightly more (~0.4GB) than its actual total: 
- `c4.large`

```
[root@login1(c4large) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-148          linux-x64       2  0.01    3.5G  157.6M     0.0     0.0
hostname              ip-10-75-0-148.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=2,exclusive=true,h_vmem=4G,h_rt=259200
```
- `c4.2xlarge`

```
[root@login1(c42xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-243          linux-x64       8  0.01   14.3G  240.8M     0.0     0.0
hostname              ip-10-75-0-243.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=8,exclusive=true,h_vmem=8G,h_rt=259200
```
- `c4.4xlarge`

```
[root@login1(c44xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-156          linux-x64      16  0.08   29.0G  371.3M     0.0     0.0
hostname              ip-10-75-0-156.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=16,exclusive=true,h_vmem=16G,h_rt=259200
```
- `c4.8xlarge`

```
[root@login1(c48xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-98           linux-x64      36  0.01   58.6G  589.2M     0.0     0.0
hostname              ip-10-75-0-98.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=36,exclusive=true,h_vmem=32G,h_rt=259200
```
- `m4.xlarge`

```
[root@login1(m4xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-220          linux-x64       4  0.02   15.3G  231.3M     0.0     0.0
hostname              ip-10-75-0-220.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=4,exclusive=true,h_vmem=8G,h_rt=259200
```
- `m4.2xlarge`

```
[root@login1(m42xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-105          linux-x64       8  0.01   31.0G  335.0M     0.0     0.0
hostname              ip-10-75-0-105.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=8,exclusive=true,h_vmem=16G,h_rt=259200
```
- `m4.4xlarge`

```
[root@login1(m44xlarge) ~]# qhost
HOSTNAME                ARCH         NCPU  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
-------------------------------------------------------------------------------
global                  -               -     -       -       -       -       -
ip-10-75-0-148          linux-x64      16  0.01   62.5G  549.7M     0.0     0.0
hostname              ip-10-75-0-148.eu-west-1.compute.internal
load_scaling          NONE
complex_values        slots=16,exclusive=true,h_vmem=32G,h_rt=259200
```

So on so fourth..

**Comment by vlj91** [comment]
Oddly in `/var/log/clusterware/cluster-sge.log`: 

```
May 26 14:23:40 Found tuple: ram_gb=2, key: ram_gb, value: 2
May 26 14:23:40 Found tuple: slots=2, key: slots, value: 2
May 26 14:23:40 Adding node: ip-10-75-0-51 (10.75.0.51) with slots=2 and vmem=2G
```

But in `qconf -me <node>`:

```
complex_values        slots=2,exclusive=true,h_vmem=4G,h_rt=259200
```

**Comment by mjtko** [comment]
The latter can be explained by this:

```
[root@login-4097f10(el7cluster) ~]# cat /opt/clusterware/etc/cluster-sge.rc  | grep min_vmem
cw_CLUSTER_SGE_min_vmem_gb=4
```

**Comment by mjtko** [comment]
The current algorithm uses a power of 2 to set the memory limits. Maybe something a little less round is in order.  Perhaps: get amount of RAM in KiB, round down to nearest GIB.

Should we need to take any account of swap where it's available or simply ignore it?

**Comment by wdm78** [comment]
Fabbo - let's ignore swap (we do on physical clusters); setting in KB, rounded down to nearest GiB gets my vote -

**Comment by mjtko** [comment]
Fixed in https://github.com/alces-software/clusterware-handlers/issues/31

**Comment by vlj91** [comment]
This is included in `2016.3`
